src/Server/MySQLHandler.cpp

```diff
@@ -601,12 +617,12 @@ static String showTableStatusReplacementQuery(const String & query)
     return query;
 }
 
-static String selectLimitReplacementQuery(const String & query)
+static std::optional<String> setSettingReplacementQuery(const String & query, const String & mysql_setting, const String & native_setting)
```

Comment:
Could we rename `native_setting` to `clickhouse_setting`? (just to remove ambiguity a bit)